traffic_portal/app/src/common/api/CDNService.js

```diff
@@ -222,7 +222,42 @@ var CDNService = function($http, locationUtils, messageModel, ENV) {
                 throw err;
 			}
 		);
-	};
+	}
+
+    this.getNotifications = function(queryParams) {
+        return $http.get(ENV.api['root'] + 'cdn_notifications', { params: queryParams }).then(
+            function(result) {
+                return result.data.response;
+            },
+            function(err) {
+                throw err;
+            }
+        );
+    };
+
+    this.createNotification = function(cdn, notification) {
+        return $http.post(ENV.api['root'] + 'cdn_notifications', { cdn: cdn.name, notification: notification}).then(
+            function(result) {
+                return result;
+            },
+            function(err) {
+                messageModel.setMessages(err.data.alerts, false);
+                throw err;
+            }
+        );
+    };
+
+    this.deleteNotification = function(cdn) {
+        return $http.delete(ENV.api['root'] + 'cdn_notifications?cdn=' + cdn.name).then(
```

Comment:
You should either pass the `cdn` parameter in the `$http.delete`'s query string parameters optional argument, or use `encodeURIComponent` to ensure proper URI component encoding.

Reply:
oh right. i meant to fix that.